test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/TrustedSignersProviderTests.cs

```diff
@@ -388,7 +411,11 @@ public void AddOrUpdateTrustedSigner_WithExistingTrustedSigner_UpdatesItSuccesfu
                 var trustedSigners = trustedSignerProvider.GetTrustedSigners();
                 trustedSigners.Should().NotBeNull();
                 trustedSigners.Count.Should().Be(3);
-                trustedSigners.Should().BeEquivalentTo(expectedTrustedSigners);
+                trustedSigners.Should().BeEquivalentTo(expectedTrustedSigners,
```

Comment:
@nkolev92 Are you familiar with this code? The line 409 above AddOrUpdateTrusterSigner will replace the existing certificate with fingerprint 'abc' with the new one of 'jkl', is that correct and the test needs to be updated to not expect the two certificates or is the code wrong?

Reply:
It is correct that `AddOrUpdateTrustedSigner(...)` replaces the certificate list for the author in this test.  The "update" here is effectively a replace operation not a merge operation.

When the scenario is performed from the command line, the end result is that the certificates are merged.  (This is correct behavior.)  This is because the CLI [gets a reference to the existing author trusted signer](https://github.com/NuGet/NuGet.Client/blob/582bcec3b291ad9fba70b64c69ccf1c830c10ad7/src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignerActionsProvider.cs#L191) and [adds the new certificate fingerprint to the existing collection of fingerprints](https://github.com/NuGet/NuGet.Client/blob/582bcec3b291ad9fba70b64c69ccf1c830c10ad7/src/NuGet.Core/NuGet.Commands/TrustedSignersCommand/TrustedSignerActionsProvider.cs#L205).  `AddOrUpdateTrustedSigner(...)` then performs a replace operation on the author.

What is clearly wrong is that this test has an incorrect expected result and that the validation didn't catch it.  Before your change the test should have failed because the updated author's certificates was different than the expected author's certificates.